cdc/sink/codec/interface.go

```diff
@@ -44,3 +49,37 @@ type EventBatchDecoder interface {
 	// NextDDLEvent returns the next DDL event if exists
 	NextDDLEvent() (*model.DDLEvent, error)
 }
+
+// Protocol is the protocol of the mq message
+type Protocol int
+
+// Enum types of the Protocol
+const (
+	ProtocolDefault Protocol = iota
+	ProtocolCanal
+)
+
+// FromString converts the protocol from string to Protocol enum type
+func (p *Protocol) FromString(protocol string) {
+	switch strings.ToLower(protocol) {
+	case "default":
+		*p = ProtocolDefault
+	case "canal":
+		*p = ProtocolCanal
+	default:
+		*p = ProtocolDefault
+		log.Warn("can't support codec protocol , using default protocol", zap.String("protocol", protocol))
+	}
+}
+
+// NewEventBatchEncoder returns a function of creating an EventBatchEncoder
+func NewEventBatchEncoder(p Protocol) func() EventBatchEncoder {
+	switch p {
+	case ProtocolDefault:
+		return NewJSONEventBatchEncoder
+	case ProtocolCanal:
+		return NewCanalEventBatchEncoder
+	default:
+		return NewJSONEventBatchEncoder
```

Comment:
I think we'd better add a warning to tell the user that the protocol of mq sink is wrong.

Reply:
fixed in https://github.com/pingcap/ticdc/pull/649/commits/c6079f5a8826846fd9c6330eda302b80b432c905